Print support for SEXPR

HanielB opened the pull request.

alex-ozdemir reviewed (changes requested):
This change is good, but we should have a unit test for it. Either a doctest in the main file, or a standalone test in [here](https://github.com/cvc5/cvc5_pythonic_api/tree/main/test).

HanielB commented:
I added the test and in the process I also added a method to create s-expression terms, which I don't think was possible before (and fixed how they should be printed, which is without the commas). I had added the printing function because these terms show up in proofs, so we need to print them.

alex-ozdemir reviewed (approved):
Great! Thanks.
